File: Software Development Module/Week9/Results.java
/*
Results.java
Billy Cussen
18/07/2020
*/

public class Results {
    
    //Data Members
    private int[][] examResults;
    private int average;
    private int min;
    private int max;
    private int averageTotal;
    private int minTotal;
    private int maxTotal;
    private String result;

    //Constructor & Setter
    public Results(int[][] examResults){
        this.examResults = examResults;
        this.average = 0;
        this.min = 0;
        this.max = 0;
        this.averageTotal = 0;
        this.minTotal = 0;
        this.maxTotal = 0;
        this.result = new String();
    }

    //Compute & Getter
    public String computeAndGetResults(){

        String resultsAverage = new String();
        this.result += "\n\n";

        for(int i = 0; i < this.examResults.length;i++){

            this.average = 0;
            this.min = this.examResults[0][0];
            this.max = this.examResults[0][0];


            for(int j = 0; j < this.examResults[0].length; j++){

                this.result += "Student "+(i+1)+", Exam "+(j+1)+": "+this.examResults[i][j]+"\n";

                this.average += this.examResults[i][j]; 

                if(this.examResults[i][j] < this.min){
                    this.min = this.examResults[i][j];
                }

                if(this.examResults[i][j] > this.max){
                    this.max = this.examResults[i][j];
                }

            }

            this.averageTotal += this.average;
            
            if(i == 0){
                this.minTotal = min;
                this.maxTotal = max;
            }

            this.average = this.average / this.examResults[0].length;
            this.result += "Student "+(i+1)+", Average Result: "+this.average+", Lowest Grade: "+this.min+", Highest Grade: "+this.max+"\n\n";

            if(min < this.minTotal){
                this.minTotal = min;
            }

            if(max > this.maxTotal){
                this.maxTotal = max;
            }

        }

        this.averageTotal = this.averageTotal / (this.examResults.length * this.examResults[0].length);
        this.result += "Average Grade of all Students: "+this.averageTotal+"\n";
        this.result += "Lowest Grade of all Students: "+this.minTotal+"\n";
        this.result += "Highest Grade of all Students: "+this.maxTotal+"\n";

        return this.result;
    }

}

File: Python Course/Software Development Module - in Python/Week9/Results.py
"""
Results.py
Billy Cussen
01/10/2020
"""

class myResults:
    def __init__(self, examResultsAll):
        self.examResultsAll = examResultsAll
        self.result = ""
        self.averageTotal = 0
        self.minTotal = 0
        self.maxTotal = 0
    
    def computeAndGetResults(self):

        for i in range(len(self.examResultsAll)):

            averageForStudent = 0
            minForStudent = self.examResultsAll[0][0]
            maxForStudent = self.examResultsAll[0][0]

            for j in range(len(self.examResultsAll[0])):
                self.result += "Student "+str(i+1)+", Exam "+str(j+1)+": "+str(self.examResultsAll[i][j])+"\n";
                averageForStudent += self.examResultsAll[i][j]
                if self.examResultsAll[i][j] < minForStudent:
                    minForStudent = self.examResultsAll[i][j]
                if self.examResultsAll[i][j] > maxForStudent:
                    maxForStudent = self.examResultsAll[i][j]
                
            self.averageTotal += averageForStudent
            if i == 0:
                self.minTotal = minForStudent
                self.maxTotal = maxForStudent

            averageForStudent = int(averageForStudent / len(self.examResultsAll[0]))
            self.result += "Student "+str(i+1)+", Average Result: "+str(averageForStudent)+", Lowest Grade: "+str(minForStudent)+", Highest Grade: "+str(maxForStudent)+"\n\n"

            if minForStudent < self.minTotal:
                self.minTotal = minForStudent
            if maxForStudent > self.maxTotal:
                self.maxTotal = maxForStudent
        
        self.averageTotal = int(self.averageTotal / (len(self.examResultsAll) * len(self.examResultsAll[0])))
        self.result += "Average Grade of all Students: "+str(self.averageTotal)+"\n";
        self.result += "Lowest Grade of all Students: "+str(self.minTotal)+"\n";
        self.result += "Highest Grade of all Students: "+str(self.maxTotal)+"\n";        
